eth/backend.go

```diff
@@ -143,8 +143,10 @@ func New(ctx *node.ServiceContext, config *Config) (*Ethereum, error) {
 
 	if !config.SkipBcVersionCheck {
 		bcVersion := rawdb.ReadDatabaseVersion(chainDb)
-		if bcVersion != core.BlockChainVersion && bcVersion != 0 {
+		if bcVersion > core.BlockChainVersion {
 			return nil, fmt.Errorf("Blockchain DB version mismatch (%d / %d).\n", bcVersion, core.BlockChainVersion)
+		} else if bcVersion < core.BlockChainVersion && bcVersion != 0 {
```

Comment:
If you leave `bcVersion != 0` in there, it means that old databases with corrupt db number (that is, 0,1,2,3 == 0) will not enter this path, and there will be no warning. 
Am I missing a reason to have it there?

Reply:
This is only printing the version. Every version will be upgraded to 4. The thing that's not allowed is to downgrade, which if I understand the code is correct.